datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -1395,4 +1398,150 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_skip_aggregation_probe_not_locked_until_skip() -> Result<()> {
```

Comment:
I verified that this test covers the change by removing the code and then this test fails like

```
thread 'aggregates::row_hash::tests::test_skip_aggregation_probe_not_locked_until_skip' (5189676) panicked at datafusion/physical-plan/src/aggregates/row_hash.rs:1538:9:
assertion `left == right` failed: Expected batch 3's rows (100) to be skipped
  left: 0
 right: 100
stack backtrace:
```